packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -120,68 +113,169 @@ enum BottomNavigationBarType {
+  /// backwards compatibility.
+  ///
+  /// The [showSelectedLabels] argument must not be non-null.
+  ///
+  /// The [showUnselectedLabels] argument defaults to `true` if [type] is
+  /// [BottomNavigationBarType.fixed] and `false` if [type] is
+  /// [BottomNavigationBarType.shifting].
   BottomNavigationBar({
     Key key,
     @required this.items,
     this.onTap,
     this.currentIndex = 0,
+    this.elevation = 8.0,
     BottomNavigationBarType type,
-    this.fixedColor,
+    Color fixedColor,
+    this.backgroundColor,
     this.iconSize = 24.0,
+    Color selectedItemColor,
+    this.unselectedItemColor,
+    this.selectedFontSize = 14.0,
+    this.unselectedFontSize = 12.0,
+    this.showSelectedLabels = true,
+    bool showUnselectedLabels,
   }) : assert(items != null),
        assert(items.length >= 2),
        assert(
         items.every((BottomNavigationBarItem item) => item.title != null) == true,
         'Every item must have a non-null title',
        ),
        assert(0 <= currentIndex && currentIndex < items.length),
-       assert(iconSize != null),
-       type = type ?? (items.length <= 3 ? BottomNavigationBarType.fixed : BottomNavigationBarType.shifting),
+       assert(elevation != null && elevation >= 0.0),
+       assert(iconSize != null && iconSize >= 0.0),
+       assert(
+         selectedItemColor != null ? fixedColor == null : true,
+         'Either selectedItemColor or fixedColor can be specified, but not both'
+       ),
+       assert(selectedFontSize != null && selectedFontSize >= 0.0),
+       assert(unselectedFontSize != null && unselectedFontSize >= 0.0),
+       assert(showSelectedLabels != null),
+       type = _type(type, items),
+       selectedItemColor = selectedItemColor ?? fixedColor,
+       showUnselectedLabels = showUnselectedLabels ?? _defaultShowUnselected(_type(type, items)),
        super(key: key);
 
-  /// The interactive items laid out within the bottom navigation bar where each item has an icon and title.
+  /// Defines the appearance of the button items that are arrayed within the
+  /// bottom navigation bar.
   final List<BottomNavigationBarItem> items;
 
-  /// The callback that is called when a item is tapped.
+  /// Called when on of the [items] is tapped.
   ///
-  /// The widget creating the bottom navigation bar needs to keep track of the
-  /// current index and call `setState` to rebuild it with the newly provided
-  /// index.
+  /// The stateful widget that creates the bottom navigation bar needs to keep
+  /// track of the index of the selected [BottomNavigationBarItem] and call
+  /// `setState` to rebuild the bottom navigation bar with the new [currentIndex].
   final ValueChanged<int> onTap;
 
-  /// The index into [items] of the current active item.
+  /// The index into [items] for the current active [BottomNavigationBarItem].
   final int currentIndex;
 
+  /// The z-coordinate of this [BottomNavigationBar].
```

Comment:
```suggestion
  /// The Z coordinate of this [BottomNavigationBar].
```

Reply:
{@macro flutter.material.material.elevation} uses "z-coordinate". Perhaps we should update the macro and any other use of z-coordinate in a separate PR

Reply:
Yeah we should probably do that, because "z-coordinate" isn't really quite right as a hyphenated expression, and the axis name is typically capitalized (although you could just as easily make an axis where they weren't).

But this is totally a nit, so feel free to ignore it too. :-)